### PR TITLE
New plugin - EasyVista

### PR DESCRIPTION
**Pull request by dsliwinski-r7**
### Description

Describe the proposed changes:

  -

### Testing

Any testing information goes here.

## PR Requirements

Developers, verify you have completed the following items by checking them off:

### Style

Review the [style guide](https://docs.rapid7.com/insightconnect/style-guide/)

- [ ] For dependencies, pin [OS package](https://docs.rapid7.com/insightconnect/style-guide/#dockerfile) and [Python package](https://docs.rapid7.com/insightconnect/style-guide/#requirements.txt) versions
- [ ] For security, set least privileged account with ``USER nobody`` in the ``Dockerfile`` when possible
- [ ] For size, use the [slim SDK images](https://docs.rapid7.com/insightconnect/sdk-guide/#sdk-guide) when possible: ``komand/python-3-37-slim-plugin`` and ``komand/python-3-37-plugin``
- [ ] For error handling, use of [PluginException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations/#plugin-exceptions) and [ConnectionTestException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations#connection-exceptions)
- [ ] For logging, use [self.logger](https://docs.rapid7.com/insightconnect/sdk-guide/#logging)
- [ ] For docs, use [changelog style](https://docs.rapid7.com/insightconnect/style-guide/#changelog)
- [ ] For docs, validate markdown with ``make validate`` which calls ``mdl`` to lint ``help.md``

### Functional Checklist
- [ ] Work fully completed
- [ ] Functional
  - [ ] Any new actions/triggers include JSON [test files](https://docs.rapid7.com/insightconnect/style-guide/#tests) in the `tests/` directory created with `icon-plugin run -c sample $action > tests/$action.json`
  - [ ] Tests should all pass unless it's a negative test. Negative tests have a naming convention of `tests/$action_bad.json`
  - [ ] Unsuccessful tests should fail by raising an exception causing the plugin to die and an object should be returned on successful test
  - [ ] Add functioning test results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -T tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -T all --debug --jq` (use PR format at end)
  - [ ] Add functioning run results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -R tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -R all --debug --jq` (use PR format at end)

### Run

<details>

```
{
  "body": {
    "log": "Connect: Connecting...\nrapid7/EasyVista:1.0.0. Step name: close_ticket\n",
    "meta": {},
    "output": {
      "result": {
        "href_hyperlink": "https://example.easyvista.com/api/v1/50005/requests/I210413_000004",
        "reference_number": "I210413_000004"
      }
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/easyvista:1.0.0 --debug run < tests/close_ticket.json
</summary>
</details>

<details>

```
{
  "body": {
    "log": "Connect: Connecting...\nrapid7/EasyVista:1.0.0. Step name: create_ticket\n",
    "meta": {},
    "output": {
      "result": {
        "href_hyperlink": "https://example.easyvista.com/api/v1/50005/requests/I210413_000012",
        "reference_number": "I210413_000012"
      }
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/easyvista:1.0.0 --debug run < tests/create_ticket.json
</summary>
</details>

<details>

```
{
  "body": {
    "log": "Connect: Connecting...\nrapid7/EasyVista:1.0.0. Step name: search_tickets\n",
    "meta": {},
    "output": {
      "results": {
        "HREF": "https://example.easyvista.com/api/v1/50005/requests?search=recipient.last_name:employee2",
        "record_count": "2",
        "records": [
          {
            "CATALOG_REQUEST": {
              "CATALOG_REQUEST_PATH": "Incidents/Test_Catalog",
              "CODE": "Test1",
              "HREF": "https://example.easyvista.com/api/v1/50005/catalog-requests/5218",
              "SD_CATALOG_ID": "5218",
              "TITLE_EN": "Test_Catalog"
            },
            "COMMENT": {
              "HREF": "https://example.easyvista.com/api/v1/50005/requests/I210408_000005/comment"
            },
            "DEPARTMENT": {
              "DEPARTMENT_CODE": "",
              "DEPARTMENT_EN": "-",
              "DEPARTMENT_ID": "5",
              "DEPARTMENT_LABEL": "",
              "DEPARTMENT_PATH": "",
              "HREF": "https://example.easyvista.com/api/v1/50005/departments/5"
            },
            "HREF": "https://example.easyvista.com/api/v1/50005/requests/I210408_000005",
            "KNOWNERROR": {
              "KNOWNERROR_PATH": "",
              "KNOWN_PROBLEMS_ID": "",
              "KP_NUMBER": "",
              "QUESTION_EN": ""
            },
            "LOCATION": {
              "CITY": "",
              "HREF": "https://example.easyvista.com/api/v1/50005/locations/6",
              "LOCATION_CODE": "",
              "LOCATION_EN": "-",
              "LOCATION_ID": "6",
              "LOCATION_PATH": ""
            },
            "MAX_RESOLUTION_DATE_UT": "",
            "RECIPIENT": {
              "BEGIN_OF_CONTRACT": "2020-12-29",
              "CELLULAR_NUMBER": " -",
              "DEPARTMENT_PATH": "",
              "EMPLOYEE_ID": "6",
              "E_MAIL": "user@example.com",
              "LAST_NAME": "employee2",
              "LOCATION_PATH": "",
              "PHONE_NUMBER": " -"
            },
            "REQUESTOR": {
              "BEGIN_OF_CONTRACT": "2020-12-29",
              "CELLULAR_NUMBER": " -",
              "DEPARTMENT_PATH": "",
              "EMPLOYEE_ID": "5",
              "E_MAIL": "user@example.com",
              "LAST_NAME": "employee1",
              "LOCATION_PATH": "",
              "PHONE_NUMBER": " -"
            },
            "REQUEST_ID": "7",
            "RFC_NUMBER": "I210408_000005",
            "STATUS": {
              "HREF": "https://example.easyvista.com/api/v1/50005/status/12",
              "STATUS_EN": "In progress",
              "STATUS_GUID": "{6E554FC1-1A5D-47E4-B303-CD98A8767AEC}",
              "STATUS_ID": "12"
            },
            "SUBMIT_DATE_UT": "2021-04-08T15:09:31.000Z"
          },
          {
            "CATALOG_REQUEST": {
              "CATALOG_REQUEST_PATH": "Incidents/Test_Catalog",
              "CODE": "Test1",
              "HREF": "https://example.easyvista.com/api/v1/50005/catalog-requests/5218",
              "SD_CATALOG_ID": "5218",
              "TITLE_EN": "Test_Catalog"
            },
            "COMMENT": {
              "HREF": "https://example.easyvista.com/api/v1/50005/requests/I210413_000002/comment"
            },
            "DEPARTMENT": {
              "DEPARTMENT_CODE": "1",
              "DEPARTMENT_EN": "Customer 1",
              "DEPARTMENT_ID": "6",
              "DEPARTMENT_LABEL": "",
              "DEPARTMENT_PATH": "Customer 1",
              "HREF": "https://example.easyvista.com/api/v1/50005/departments/6"
            },
            "HREF": "https://example.easyvista.com/api/v1/50005/requests/I210413_000002",
            "KNOWNERROR": {
              "KNOWNERROR_PATH": "",
              "KNOWN_PROBLEMS_ID": "",
              "KP_NUMBER": "",
              "QUESTION_EN": ""
            },
            "LOCATION": {
              "CITY": "",
              "HREF": "https://example.easyvista.com/api/v1/50005/locations/6",
              "LOCATION_CODE": "",
              "LOCATION_EN": "-",
              "LOCATION_ID": "6",
              "LOCATION_PATH": ""
            },
            "MAX_RESOLUTION_DATE_UT": "",
            "RECIPIENT": {
              "BEGIN_OF_CONTRACT": "2020-12-29",
              "CELLULAR_NUMBER": " -",
              "DEPARTMENT_PATH": "",
              "EMPLOYEE_ID": "6",
              "E_MAIL": "user@example.com",
              "LAST_NAME": "employee2",
              "LOCATION_PATH": "",
              "PHONE_NUMBER": " -"
            },
            "REQUESTOR": {
              "BEGIN_OF_CONTRACT": "2020-12-29",
              "CELLULAR_NUMBER": " -",
              "DEPARTMENT_PATH": "",
              "EMPLOYEE_ID": "11",
              "E_MAIL": "user@example.com",
              "LAST_NAME": "employee7",
              "LOCATION_PATH": "",
              "PHONE_NUMBER": " -"
            },
            "REQUEST_ID": "27",
            "RFC_NUMBER": "I210413_000002",
            "STATUS": {
              "HREF": "https://example.easyvista.com/api/v1/50005/status/12",
              "STATUS_EN": "In progress",
              "STATUS_GUID": "{6E554FC1-1A5D-47E4-B303-CD98A8767AEC}",
              "STATUS_ID": "12"
            },
            "SUBMIT_DATE_UT": "2021-04-13T15:20:52.000Z"
          }
        ],
        "total_record_count": "2"
      }
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/easyvista:1.0.0 --debug run < tests/search_tickets_recipient_name.json
</summary>
</details>

<details>

```
{
  "body": {
    "log": "Connect: Connecting...\nrapid7/EasyVista:1.0.0. Step name: search_tickets\n",
    "meta": {},
    "output": {
      "results": {
        "HREF": "https://example.easyvista.com.com/api/v1/50005/requests?search=rfc_number:I210413_000011",
        "record_count": "1",
        "records": [
          {
            "CATALOG_REQUEST": {
              "CATALOG_REQUEST_PATH": "Incidents/Test_Catalog",
              "CODE": "Test1",
              "HREF": "https://example.easyvista.com/api/v1/50005/catalog-requests/5218",
              "SD_CATALOG_ID": "5218",
              "TITLE_EN": "Test_Catalog"
            },
            "COMMENT": {
              "HREF": "https://example.easyvista.com/api/v1/50005/requests/I210413_000011/comment"
            },
            "DEPARTMENT": {
              "DEPARTMENT_CODE": "1",
              "DEPARTMENT_EN": "Customer 1",
              "DEPARTMENT_ID": "6",
              "DEPARTMENT_LABEL": "",
              "DEPARTMENT_PATH": "Customer 1",
              "HREF": "https://example.easyvista.com/api/v1/50005/departments/6"
            },
            "HREF": "https://example.easyvista.com/api/v1/50005/requests/I210413_000011",
            "KNOWNERROR": {
              "KNOWNERROR_PATH": "",
              "KNOWN_PROBLEMS_ID": "",
              "KP_NUMBER": "",
              "QUESTION_EN": ""
            },
            "LOCATION": {
              "CITY": "",
              "HREF": "https://example.easyvista.com/api/v1/50005/locations/6",
              "LOCATION_CODE": "",
              "LOCATION_EN": "-",
              "LOCATION_ID": "6",
              "LOCATION_PATH": ""
            },
            "MAX_RESOLUTION_DATE_UT": "",
            "RECIPIENT": {
              "BEGIN_OF_CONTRACT": "2020-12-29",
              "CELLULAR_NUMBER": " -",
              "DEPARTMENT_PATH": "",
              "EMPLOYEE_ID": "7",
              "E_MAIL": "user@example.com",
              "LAST_NAME": "employee3",
              "LOCATION_PATH": "",
              "PHONE_NUMBER": " -"
            },
            "REQUESTOR": {
              "BEGIN_OF_CONTRACT": "2020-12-29",
              "CELLULAR_NUMBER": " -",
              "DEPARTMENT_PATH": "",
              "EMPLOYEE_ID": "11",
              "E_MAIL": "user@example.com",
              "LAST_NAME": "employee7",
              "LOCATION_PATH": "",
              "PHONE_NUMBER": " -"
            },
            "REQUEST_ID": "36",
            "RFC_NUMBER": "I210413_000011",
            "STATUS": {
              "HREF": "https://example.easyvista.com/api/v1/50005/status/12",
              "STATUS_EN": "In progress",
              "STATUS_GUID": "{6E554FC1-1A5D-47E4-B303-CD98A8767AEC}",
              "STATUS_ID": "12"
            },
            "SUBMIT_DATE_UT": "2021-04-13T17:04:33.000Z"
          }
        ],
        "total_record_count": "1"
      }
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/easyvista:1.0.0 --debug run < tests/search_tickets_rfc_number.json
</summary>
</details>

<details>

```
{
  "body": {
    "log": "Connect: Connecting...\nrapid7/EasyVista:1.0.0. Step name: search_tickets\n",
    "meta": {},
    "output": {
      "results": {
        "HREF": "https://example.easyvista.com/api/v1/50005/requests?search=status.status_en:Closed",
        "record_count": "2",
        "records": [
          {
            "CATALOG_REQUEST": {
              "CATALOG_REQUEST_PATH": "Incidents/Test_Catalog",
              "CODE": "Test1",
              "HREF": "https://example.easyvista.com/api/v1/50005/catalog-requests/5218",
              "SD_CATALOG_ID": "5218",
              "TITLE_EN": "Test_Catalog"
            },
            "COMMENT": {
              "HREF": "https://example.easyvista.com/api/v1/50005/requests/I210409_000002/comment"
            },
            "DEPARTMENT": {
              "DEPARTMENT_CODE": "",
              "DEPARTMENT_EN": "-",
              "DEPARTMENT_ID": "5",
              "DEPARTMENT_LABEL": "",
              "DEPARTMENT_PATH": "",
              "HREF": "https://example.easyvista.com/api/v1/50005/departments/5"
            },
            "HREF": "https://example.easyvista.com/api/v1/50005/requests/I210409_000002",
            "KNOWNERROR": {
              "KNOWNERROR_PATH": "",
              "KNOWN_PROBLEMS_ID": "",
              "KP_NUMBER": "",
              "QUESTION_EN": ""
            },
            "LOCATION": {
              "CITY": "",
              "HREF": "https://example.easyvista.com/api/v1/50005/locations/6",
              "LOCATION_CODE": "",
              "LOCATION_EN": "-",
              "LOCATION_ID": "6",
              "LOCATION_PATH": ""
            },
            "MAX_RESOLUTION_DATE_UT": "",
            "RECIPIENT": {
              "BEGIN_OF_CONTRACT": "2021-04-02",
              "CELLULAR_NUMBER": " -",
              "DEPARTMENT_PATH": "",
              "EMPLOYEE_ID": "16",
              "E_MAIL": "user@example.com",
              "LAST_NAME": "Admin",
              "LOCATION_PATH": "",
              "PHONE_NUMBER": " -"
            },
            "REQUESTOR": {
              "BEGIN_OF_CONTRACT": "2021-04-02",
              "CELLULAR_NUMBER": " -",
              "DEPARTMENT_PATH": "",
              "EMPLOYEE_ID": "16",
              "E_MAIL": "user@example.com",
              "LAST_NAME": "Admin",
              "LOCATION_PATH": "",
              "PHONE_NUMBER": " -"
            },
            "REQUEST_ID": "9",
            "RFC_NUMBER": "I210409_000002",
            "STATUS": {
              "HREF": "https://example.easyvista.com/api/v1/50005/status/8",
              "STATUS_EN": "Closed",
              "STATUS_GUID": "{C3D9DFA7-7A21-46C2-B3A3-8BC50C9FF4F3}",
              "STATUS_ID": "8"
            },
            "SUBMIT_DATE_UT": "2021-04-09T11:29:36.947Z"
          },
          {
            "CATALOG_REQUEST": {
              "CATALOG_REQUEST_PATH": "Incidents/Test_Catalog",
              "CODE": "Test1",
              "HREF": "https://example.easyvista.com/api/v1/50005/catalog-requests/5218",
              "SD_CATALOG_ID": "5218",
              "TITLE_EN": "Test_Catalog"
            },
            "COMMENT": {
              "HREF": "https://example.easyvista.com/api/v1/50005/requests/I210409_000007/comment"
            },
            "DEPARTMENT": {
              "DEPARTMENT_CODE": "",
              "DEPARTMENT_EN": "-",
              "DEPARTMENT_ID": "5",
              "DEPARTMENT_LABEL": "",
              "DEPARTMENT_PATH": "",
              "HREF": "https://example.easyvista.com/api/v1/50005/departments/5"
            },
            "HREF": "https://example.easyvista.com/api/v1/50005/requests/I210409_000007",
            "KNOWNERROR": {
              "KNOWNERROR_PATH": "",
              "KNOWN_PROBLEMS_ID": "",
              "KP_NUMBER": "",
              "QUESTION_EN": ""
            },
            "LOCATION": {
              "CITY": "",
              "HREF": "https://example.easyvista.com/api/v1/50005/locations/6",
              "LOCATION_CODE": "",
              "LOCATION_EN": "-",
              "LOCATION_ID": "6",
              "LOCATION_PATH": ""
            },
            "MAX_RESOLUTION_DATE_UT": "",
            "RECIPIENT": {
              "BEGIN_OF_CONTRACT": "2020-12-29",
              "CELLULAR_NUMBER": " -",
              "DEPARTMENT_PATH": "",
              "EMPLOYEE_ID": "10",
              "E_MAIL": "user@example.com",
              "LAST_NAME": "employee6",
              "LOCATION_PATH": "",
              "PHONE_NUMBER": " -"
            },
            "REQUESTOR": {
              "BEGIN_OF_CONTRACT": "2020-12-29",
              "CELLULAR_NUMBER": " -",
              "DEPARTMENT_PATH": "",
              "EMPLOYEE_ID": "13",
              "E_MAIL": "user@example.com",
              "LAST_NAME": "employee9",
              "LOCATION_PATH": "",
              "PHONE_NUMBER": " -"
            },
            "REQUEST_ID": "14",
            "RFC_NUMBER": "I210409_000007",
            "STATUS": {
              "HREF": "https://example.easyvista.com/api/v1/50005/status/8",
              "STATUS_EN": "Closed",
              "STATUS_GUID": "{C3D9DFA7-7A21-46C2-B3A3-8BC50C9FF4F3}",
              "STATUS_ID": "8"
            },
            "SUBMIT_DATE_UT": "2021-04-09T13:48:55.000Z"
          }
        ],
        "total_record_count": "2"
      }
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/easyvista:1.0.0 --debug run < tests/search_tickets_status.json
</summary>
</details>

<details>

```
{
  "body": {
    "log": "Connect: Connecting...\nrapid7/EasyVista:1.0.0. Step name: update_ticket\n",
    "meta": {},
    "output": {
      "result": {
        "href_hyperlink": "https://example.easyvista.com/api/v1/50005/requests/I210413_000006",
        "reference_number": "I210413_000006"
      }
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/easyvista:1.0.0 --debug run < tests/update_ticket.json
</summary>
</details>

<details>

```
[*] Validating plugin with all validators at .

[*] Running Integration Validators...
[*] Executing validator HelpValidator
[*] Executing validator ChangelogValidator
[*] Executing validator CloudReadyConnectionCredentialTokenValidator
[*] Executing validator RequiredKeysValidator
[*] Executing validator UseCaseValidator
[*] Executing validator SpecPropertiesValidator
[*] Executing validator SpecVersionValidator
[*] Executing validator FilesValidator
[*] Executing validator TagValidator
[*] Executing validator DescriptionValidator
[*] Executing validator TitleValidator
[*] Executing validator VendorValidator
[*] Executing validator DefaultValueValidator
[*] Executing validator IconValidator
[*] Executing validator RequiredValidator
[*] Executing validator VersionValidator
[*] Executing validator DockerfileParentValidator
[*] Executing validator LoggingValidator
[*] Executing validator ProfanityValidator
[*] Executing validator AcronymValidator
[*] Executing validator JSONValidator
[*] Executing validator OutputValidator
[*] Executing validator RegenerationValidator
[*] Executing validator HelpInputOutputValidator
[*] Executing validator SupportValidator
[*] Executing validator RuntimeValidator
[*] Executing validator VersionPinValidator
[*] Executing validator EncodingValidator
[*] Executing validator ExampleInputValidator
[*] Executing validator ExceptionValidator
[*] Executing validator CredentialsValidator
[*] Executing validator PasswordValidator
[*] Executing validator PrintValidator
[*] Executing validator ConfidentialValidator
[*] Executing validator DockerValidator
[*] Executing validator URLValidator
WARNING: URLs found that return a 4xx code. Verify they are publicly accessible and if not, update with a working URL.
violation: plugin.spec.yaml[303]: https://example.easyvista.com
violation: plugin.spec.yaml[306]: https://example.easyvista.com
violation: help.md[76]: https://example.easyvista.com/api/v1...
violation: help.md[81]: https://example.easyvista.com/api/v1...
violation: help.md[98]: https://example.easyvista.com/api/v1...
violation: help.md[115]: https://example.easyvista.com/api/v1...
violation: help.md[119]: https://example.easyvista.com/api/v1...
violation: help.md[132]: https://example.easyvista.com/api/v1...
violation: help.md[141]: https://example.easyvista.com/api/v1...
violation: help.md[185]: https://example.easyvista.com/api/v1...
violation: help.md[272]: https://example.easyvista.com/api/v1...
violation: help.md[361]: https://example.easyvista.com/api/v1...
[*] Plugin failed validation! The following validation errors occurred:

Validator "IconValidator" failed! 
	Cause: icon.png file not included in plugin.


----
[*] Total time elapsed: 16526.549ms
```

<summary>
icon-validate --all .
</summary>
</details>

<details>

```
[*] Use ``make menu`` for available targets
[*] Including available Makefiles: ../tools/Makefiles/Helpers.mk ../tools/Makefiles/Colors.mk
--
[*] Running validators
[*] Validating plugin at .

[*] Running Integration Validators...
[*] Executing validator HelpValidator
[*] Executing validator ChangelogValidator
[*] Executing validator CloudReadyConnectionCredentialTokenValidator
[*] Executing validator RequiredKeysValidator
[*] Executing validator UseCaseValidator
[*] Executing validator SpecPropertiesValidator
[*] Executing validator SpecVersionValidator
[*] Executing validator FilesValidator
[*] Executing validator TagValidator
[*] Executing validator DescriptionValidator
[*] Executing validator TitleValidator
[*] Executing validator VendorValidator
[*] Executing validator DefaultValueValidator
[*] Executing validator IconValidator
[*] Executing validator RequiredValidator
[*] Executing validator VersionValidator
[*] Executing validator DockerfileParentValidator
[*] Executing validator LoggingValidator
[*] Executing validator ProfanityValidator
[*] Executing validator AcronymValidator
[*] Executing validator JSONValidator
[*] Executing validator OutputValidator
[*] Executing validator RegenerationValidator
[*] Executing validator HelpInputOutputValidator
[*] Executing validator SupportValidator
[*] Executing validator RuntimeValidator
[*] Executing validator VersionPinValidator
[*] Executing validator EncodingValidator
[*] Executing validator ExampleInputValidator
[*] Plugin failed validation! The following validation errors occurred:

Validator "IconValidator" failed! 
	Cause: icon.png file not included in plugin.


----
[*] Total time elapsed: 1743.357ms

[*] Validating spec with js-yaml
[SUCCESS] Passes js-yaml spec check


[*] For Markdown linting, install mdl: gem install mdl

[*] Validating python files for style...
[SUCCESS] Passes flake8 linting

[*] Validating python files for security vulnerabilities...
[main]	INFO	profile include tests: None
[main]	INFO	profile exclude tests: None
[main]	INFO	cli include tests: None
[main]	INFO	cli exclude tests: None
[main]	INFO	running on Python 3.7.3
Run started:2021-04-15 18:43:39.526914

Test results:
	No issues identified.

Code scanned:
	Total lines of code: 2068
	Total lines skipped (#nosec): 0

Run metrics:
	Total issues (by severity):
		Undefined: 0.0
		Low: 0.0
		Medium: 0.0
		High: 0.0
	Total issues (by confidence):
		Undefined: 0.0
		Low: 0.0
		Medium: 0.0
		High: 0.0
Files skipped (0):
[SUCCESS] Passes bandit security checks

[*] For spell checking, install misspell: go get -u github.com/client9/misspell/cmd/misspell
```

<summary>
make validate
</summary>
</details>

### CODE REVIEW

File: easyvista/plugin.spec.yaml [hunk 1]
```diff
@@ -0,0 +1,733 @@
...
+        example: 1
+      ci_serial:
+        title: CI Serial
+        description: Serial number of the Configuration Item
+        type: string
+        required: false
+        example: KD78QGJYU
+      comment:
+        title: Comment
+        description: Comment explaining the updating of the ticket
```
Comment (jmcadams-r7) [inline easyvista/plugin.spec.yaml:553]:
```suggestion
        description: Comment that explains the reason for the update
```

File: easyvista/plugin.spec.yaml [hunk 2]
```diff
@@ -0,0 +1,733 @@
...
+        example: CP01
+      description:
+        title: Description
+        description: Description of the ticket
+        type: string
+        required: false
+        example: Example description
+      external_reference:
+        title: External Reference
+        description: Identifier of the object attributed by an external application
```
Comment (jmcadams-r7) [inline easyvista/plugin.spec.yaml:571]:
```suggestion
        description: Identifier of the object used by an external application
```

File: easyvista/plugin.spec.yaml [hunk 3]
```diff
@@ -0,0 +1,733 @@
...
+        example: I210412_000001
+      catalog_guid:
+        title: Catalog GUID
+        description: Identifier of the topic of the ticket. Required if the ticket needs to be requalified before closing
+        type: string
+        required: false
+        example: 44D88612-FEA8-A8F3-6DE8-2E1278ABB02F
+      comment:
+        title: Comment
+        description: Comment explaining the closing of the ticket
```
Comment (jmcadams-r7) [inline easyvista/plugin.spec.yaml:689]:
```suggestion
        description: Comment that explains why the ticket was closed.
```

**Review by jmcadams-r7 — CHANGES_REQUESTED** [review]
Really good code...just some minor nit picks on descriptions.

File: easyvista/icon_easyvista/util/api.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+class EasyVistaApi:
+    def __init__(self, client_login: dict, account: int, hostname: str):
+        self.base_url = f"{hostname}/api/v1/{account}/"
+        self.username = client_login.get("username")
+        self.password = client_login.get("password")
+
+    def _call_api(self, method: str, endpoint: str, json: dict = None):
+
+        response = requests.request(
+            url=self.base_url + endpoint,
```
Comment (jmcadams-r7) [inline easyvista/icon_easyvista/util/api.py:16]: You don't need to fix this, but next time you might want to use

```urllib.parse.urljoin(self.base_url, endpoint)```

Then you don't have to be careful with the trailing or leading `/ `

File: easyvista/plugin.spec.yaml [hunk 5]
```diff
@@ -0,0 +1,733 @@
...
+        example: 1
+    output:
+      result:
+        title: Result
+        description: Result that includes URL link (HREF) and reference number of the updated ticket
+        type: ticket_data
+        required: true
+  close_ticket:
+    title: Close Ticket
+    description: Close the EasyVista ticket
```
Comment (mrinehart-r7) [inline easyvista/plugin.spec.yaml:673]:
```suggestion
    description: Close an EasyVista ticket
```

File: easyvista/plugin.spec.yaml [hunk 6]
```diff
@@ -0,0 +1,733 @@
...
+        example: DC97DD1D-0F35-4153-B0E1-0F2E0155365D
+    output:
+      result:
+        title: Result
+        description: Result that includes URL link (HREF) and reference number of the closed ticket
+        type: ticket_data
+        required: true
+  search_tickets:
+    title: Search Tickets
+    description: Search the EasyVista tickets
```
Comment (mrinehart-r7) [inline easyvista/plugin.spec.yaml:720]:
```suggestion
    description: Search for EasyVista tickets
```

File: easyvista/plugin.spec.yaml [hunk 7]
```diff
@@ -0,0 +1,733 @@
...
+    input:
+      query:
+        title: Query
+        description: Search query, if empty, returns all tickets
```
Comment (mrinehart-r7) [inline easyvista/plugin.spec.yaml:724]:
```suggestion
        description: Search query. Returns all tickets if left empty
```

File: easyvista/help.md [hunk 8]
```diff
@@ -0,0 +1,488 @@
...
+}
+```
+
+## Technical Details
+
+### Actions
+
+#### Search Tickets
+
+This action is used to search the EasyVista tickets.
```
Comment (mrinehart-r7) [inline easyvista/help.md:45]: Can you include a link to the search filters with this? https://wiki.easyvista.com/xwiki/bin/view/Documentation/REST+API+-+Options+for+Fields#SearchFilterOptions

**Review by mrinehart-r7 — CHANGES_REQUESTED** [review]

File: easyvista/plugin.spec.yaml [hunk 9]
```diff
@@ -0,0 +1,733 @@
...
+      type: string
+      required: false
+
+connection:
+  hostname:
+    title: EasyVista Server Hostname
+    description: The hostname of your EasyVista server, e.g. https://example.easyvista.com
+    type: string
+    required: true
+    example: https://example.easyvista.com
```
Comment (jschipp-r7) [inline easyvista/plugin.spec.yaml:306]: If a user provides a trailing path or extra space or newline we should strip it.

Sometimes users will copy/paste from browser and not think about it:

`https://example.easyvista.com/home`
`https://example.easyvista.com    `

File: easyvista/plugin.spec.yaml [hunk 10]
```diff
@@ -0,0 +1,733 @@
...
+      type: string
+      required: false
+    reference_number:
+      title: Reference Number
+      description: Reference number of the ticket
+      type: string
+      required: false
+
+connection:
+  hostname:
```
Comment (jschipp-r7) [inline easyvista/plugin.spec.yaml:301]: The example looks like a URL rather than a hostname. If so, we should rename this to `url` and update the description and title.

**Review by jmcadams-r7 — APPROVED** [review]

**Review by mrinehart-r7 — APPROVED** [review]
